src/components/vice/admin/filter/efcs.js

```diff
@@ -0,0 +1,265 @@
+import { JSONPath } from "jsonpath-plus";
+
+class ExtractFilterCompare {
+    constructor(extractorFn, filterFn) {
+        this.extractor = extractorFn;
+        this.filterFn = filterFn;
+    }
+
+    extractIt(fromObject, ...args) {
+        const extractorString = this.extractor(...args);
+        return JSONPath({ json: fromObject, path: extractorString });
+    }
+
+    filterIt(fromObject, ...args) {
+        const filterString = this.filterFn(...args);
+
+        return JSONPath({ json: fromObject, path: filterString });
+    }
+
+    compareIt(one, two) {
+        return one === two;
+    }
+}
+
+class DeploymentsEFC extends ExtractFilterCompare {
+    filterIt(fromObject, ...args) {
+        const filterString = this.filterFn(...args);
+        const copy = {
+            ...fromObject,
+        };
+        copy.deployments = JSONPath({
+            json: copy,
+            path: filterString,
+        });
+        return copy;
+    }
+}
+
+const deployments = {
+    image: new DeploymentsEFC(
+        () => "$.deployments[*].image",
+        (image) => `$.deployments[?(@.image=='${image}')]`
+    ),
```

Comment:
Since I'm looking at it myself as I sit at a car dealership: I think most of this syntax (the `"$.` strings, at least) is from https://www.npmjs.com/package/jsonpath-plus